### ISSUE TITLE
Make an object classifier for chess pawns

### ISSUE THREAD

**Issue by abhidxt299** [abhidxt299/RoManOV_Automation#34]
Select a suitable object detector and train it on a dataset of chess pawns. You can use any of the dataset in the links provided.

1. https://public.roboflow.ai/object-detection/chess-full
2. 
https://www.kaggle.com/koryakinp/chess-positions

**Comment by DevanshD3** [comment]
Bhaiya is this the same issue that was given in the Whatsapp group? Like we had to train the model with different classes of chess pawns.

**Comment by abhidxt299** [comment]
Yes, it is. But this time, I wish you'd try out a different method of classifying the object. You can first try to implement whatever is shown in the 1st link.

**Comment by afrajshaikh** [comment]
I want to work on this isue.

**Comment by afrajshaikh** [comment]
Is there any specific subtask I need to do or the whole task. And do we need to choose an object detector directly from Roboflow?

**Comment by abhidxt299** [comment]
You have to do the entire task. For the 2nd part of the question, it's totally up to you. But I'd suggest you to start with Roboflow.

**Comment by afrajshaikh** [comment]
Ok

**Comment by Raghav-Dhir** [comment]
hey @afrajshaikh anyway we can work on this together, I was also interested , was thinking of using rcnn

**Comment by afrajshaikh** [comment]
hi @Raghav-Dhir, what about YOLO v5 ?

**Comment by Raghav-Dhir** [comment]
i asked @abhidxt299 about that , he said it would be preferred if we go for R-CNN's or RFCN's.

**Comment by afrajshaikh** [comment]
Cool

**Comment by afrajshaikh** [comment]
Have decided which to use or still thinking about that?

**Comment by afrajshaikh** [comment]
@Raghav-Dhir have you decided which one to use?

**Comment by Raghav-Dhir** [comment]
https://towardsdatascience.com/step-by-step-r-cnn-implementation-from-scratch-in-python-e97101ccde55
i think this should be good starting point

**Comment by afrajshaikh** [comment]
Cool

**Comment by afrajshaikh** [comment]
@Raghav-Dhir in which format you downloaded the dataset?

**Comment by Raghav-Dhir** [comment]
in csv format and also i think we should have a common collab file

**Comment by Raghav-Dhir** [comment]
although i have still figure out some things 
https://colab.research.google.com/drive/1zeN3JqbS6jDEWxa35RMJ-5Z5OndJ2qM-?usp=sharing
we can do changes in this

**Comment by afrajshaikh** [comment]
I have requested access

**Comment by afrajshaikh** [comment]
@Raghav-Dhir i talked to @abhidxt299 he said we dont need to implement a detector from scratch right now, we can directly select a detector for now as we will be making our own afterwards. There are many object detectors available on roboflow under models section you check that

**Comment by abhidxt299** [comment]
@Raghav-Dhir @afrajshaikh you can try using YOLOv3 model. It'd a really good start.

**Comment by Raghav-Dhir** [comment]
Sure

**Comment by Raghav-Dhir** [comment]
have made some changes in the shared notebook. wanted you to check before training the model

**Comment by afrajshaikh** [comment]
@Raghav-Dhir which model are you training?

**Comment by afrajshaikh** [comment]
I just kept YOLO v3 for training

**Comment by afrajshaikh** [comment]
like half an hour ago

**Comment by Raghav-Dhir** [comment]
yeah it was around 500 epochs so i reduced that 
and added the _classes.txt and _annotations.txt

**Comment by Raghav-Dhir** [comment]
just see the file once

**Comment by Raghav-Dhir** [comment]
also tell me if you can see the keras-yolo3 folder because i am not sure if it will visible to you or not

**Comment by afrajshaikh** [comment]
@Raghav-Dhir i guess you first downloaded the data on desktop and then you are importing it, you could have just imported the dataset by copying and pasting the download link.

**Comment by afrajshaikh** [comment]
I am not sure about the import dataset (from dektop) code, i tried it previously but was unable to do it, I think for using it in collab you need to first upload it on google drive folder. Seniors plz confirm.

**Comment by Raghav-Dhir** [comment]
@afrajshaikh i have copied the download link only

**Comment by afrajshaikh** [comment]
when i imported the dateset annotation was in txt format only, in your case it is in csv format so you converted it into txt right?

**Comment by afrajshaikh** [comment]
It has to be in txt then you can train it.

**Comment by Raghav-Dhir** [comment]
yes i did

**Comment by afrajshaikh** [comment]
For how many epoch you are training and your videos.py code block took lot of time?

**Comment by afrajshaikh** [comment]
https://drive.google.com/file/d/1-8tZgdZrPOkF9nHhOQL7IoNpTVS--H_G/view?usp=sharing    I trained the model by keeping epochs 200. For training batch size was 8 and while testing it was 4. Test set loss was around 43 and validation set loss was around 40.

**Comment by afrajshaikh** [comment]
validation set batch_size=4*
